src/Benchmarks/Benchmarks/DeserializeFromJsonBodyBenchmark.cs

```diff
@@ -0,0 +1,48 @@
+using BenchmarkDotNet.Attributes;
+using GraphQL.Introspection;
+using GraphQL.Server.Common;
+using Microsoft.AspNetCore.Http;
+using GraphQL.Server;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using NsjDeserializer = GraphQL.Server.Transports.AspNetCore.NewtonsoftJson.GraphQLRequestDeserializer;
+using StjDeserializer = GraphQL.Server.Transports.AspNetCore.SystemTextJson.GraphQLRequestDeserializer;
+using GraphQL.Server.Transports.AspNetCore.Common;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace GraphQL.Server.Benchmarks
+{
+    [MemoryDiagnoser]
+    [RPlotExporter, CsvMeasurementsExporter]
+    public class DeserializeFromJsonBodyBenchmark
+    {
+        private NsjDeserializer _nsjDeserializer;
+        private StjDeserializer _stjDeserializer;
+        private HttpRequest _httpRequest;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        //public DeserializeFromJsonBodyBenchmark()
+        {
+            _nsjDeserializer = new NsjDeserializer(s => { });
+            _stjDeserializer = new StjDeserializer(s => { });
+
+            var gqlRequest = new GraphQLRequest { Query = SchemaIntrospection.IntrospectionQuery };
+            var gqlRequestJson = Serializer.ToJson(gqlRequest);
+            var httpContext = new DefaultHttpContext();
+            httpContext.Features.Set<IRequestBodyPipeFeature>(new RequestBodyPipeFeature(httpContext));
+            httpContext.Request.Body = new MemoryStream(Encoding.UTF8.GetBytes(gqlRequestJson));
```

Comment:
Don't you need to reset the position of this stream before starting each test?

Reply:
After switching to preview-1505:

``` ini

BenchmarkDotNet=v0.12.0, OS=Windows 10.0.17134.1304 (1803/April2018Update/Redstone4)
Intel Core i5-6200U CPU 2.30GHz (Skylake), 1 CPU, 4 logical and 2 physical cores
Frequency=2343753 Hz, Resolution=426.6661 ns, Timer=TSC
.NET Core SDK=3.1.101
  [Host]     : .NET Core 3.1.1 (CoreCLR 4.700.19.60701, CoreFX 4.700.19.60801), X86 RyuJIT
  DefaultJob : .NET Core 3.1.1 (CoreCLR 4.700.19.60701, CoreFX 4.700.19.60801), X86 RyuJIT


```
|         Method |        Mean |     Error |    StdDev |  Ratio | RatioSD |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|--------------- |------------:|----------:|----------:|-------:|--------:|-------:|------:|------:|----------:|
| NewtonsoftJson |    358.4 ns |   7.39 ns |   6.92 ns |   1.00 |    0.00 | 2.1553 |     - |     - |   3.31 KB |
| SystemTextJson | 88,478.9 ns | 351.83 ns | 329.11 ns | 246.92 |    4.75 | 0.8545 |     - |     - |   1.45 KB |

So I think the problem is shared state ans stream position.

Reply:
I thought that the GlobalSetup ran every benchmark, but I guess the class isn't instantiated every benchmark, so yea, probably need to reset the stream pos.

Reply:
Yep, needed resetting. Will be fixed in next push.